theseus/extlib/test_mat_mult.py

```diff
@@ -0,0 +1,119 @@
+import pytest  # noqa: F401
+import torch  # needed for import of Torch C++ extensions to work
+from scipy.sparse import csr_matrix
+
+from theseus.extlib.sparse_matrix_utils import generate_sparse_matrix
+
+
+def check_mat_mult(batch_size, rows, cols, fill, verbose=False):
+    if not torch.cuda.is_available():
+        return
+    from theseus.extlib.mat_mult import apply_damping, mat_vec, mult_MtM, tmat_vec
```

Comment:
Same as pervious PR, adding import stuff to root init file.

Reply:
are we sure we want those on the root init file? those are really low level operations (and not putting them in the root file does not prevent an experienced user from importing them, they are just less advertised and the user should be aware of what she's doing)

Reply:
Makes sense. Do you or @luisenp think it would be worth pulling any of these to root init? We can always do this in the future after we see a constant need for any of them.

Reply:
Given the discussion in PR 22, we will leave as is for now.

Reply:
Agree with @maurimo here. Another possibility down the line is to have some sort of SparseMatrix class and have these be overridden operators.